src/test/rpc_tests.cpp

```diff
@@ -214,6 +217,32 @@ BOOST_AUTO_TEST_CASE(rpc_parse_monetary_values)
     BOOST_CHECK_THROW(AmountFromValue(ValueFromString("1e+11")), UniValue); //overflow error
     BOOST_CHECK_THROW(AmountFromValue(ValueFromString("1e11")), UniValue); //overflow error signless
     BOOST_CHECK_THROW(AmountFromValue(ValueFromString("93e+9")), UniValue); //overflow error
+
+    // values rejected by ParseFixedPoint raise invalid amount
+    BOOST_CHECK_THROW(AmountFromValue(ValueFromString("0.000000001")), UniValue);
+    BOOST_CHECK_THROW(AmountFromValue(ValueFromString("1.111111111")), UniValue);
+    BOOST_CHECK_THROW(AmountFromValue(ValueFromString("11111111111")), UniValue);
+
+    // Test FeeRateFromValueInSatB
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("0")) == CFeeRate{0});
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("0.00000000")) == CFeeRate{0});
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("0.00000000")).IsZero());
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("0.001")) == CFeeRate{1});
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("0.00100000")) == CFeeRate{1});
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("1")) == CFeeRate{1000});
+    BOOST_CHECK(!FeeRateFromValueInSatB(ValueFromString("1")).IsZero());
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("1.000")) == CFeeRate{1000});
+    BOOST_CHECK(FeeRateFromValueInSatB(ValueFromString("1.2340000")) == CFeeRate{1234});
+
+    // values not representable by CFeeRate raise invalid amount
+    BOOST_CHECK_THROW(FeeRateFromValueInSatB(ValueFromString("0.00099999")), UniValue);
+    BOOST_CHECK_THROW(FeeRateFromValueInSatB(ValueFromString("0.00000001")), UniValue);
+
+    // values rejected by ParseFixedPoint raise invalid amount
+    BOOST_CHECK_THROW(FeeRateFromValueInSatB(ValueFromString("0.000999999")), UniValue);
+    BOOST_CHECK_THROW(FeeRateFromValueInSatB(ValueFromString("0.000000001")), UniValue);
```

Comment:
How far do we want to reach with these tests. This is basically "testing" that `FeeRateFromValueInSatB()` calls `AmountFromValue()` calls `ParseFixedPoint()` & `throw`s. Which is the same as:
```cpp
BOOST_CHECK_THROW(AmountFromValue(ValueFromString("0.000000001")), UniValue);
```
above, with an additional function call to get to `throw`.

However, because nothing specific is being checked, you could imagine how `FeeRateFromValueInSatB()` may end up being refactored such that there is some other throwing call, i.e:
```diff
diff --git a/src/rpc/util.cpp b/src/rpc/util.cpp
index 1a83b9222..02604b595 100644
--- a/src/rpc/util.cpp
+++ b/src/rpc/util.cpp
@@ -89,6 +89,7 @@ CAmount AmountFromValue(const UniValue& value)
 
 CFeeRate FeeRateFromValueInSatB(const UniValue& value)
 {
+    this_might_throw();
     const CAmount amount{AmountFromValue(value)};
     const CFeeRate fee_rate{CFeeRate::FromSatB(amount)};
```

and the tests still pass, but only because `this_might_throw()` throws for some reason. Now what are we testing? This is also why I'm not a fan of the `raise invalid amount` type comments. That might be what we want the code to be testing, but that doesn't mean that's what it's doing.

Reply:
Relooking at it in light of your thoughts here.

Reply:
Dropped these unit tests and comments and combined the tests into the same commit as the function creation and the codebase updates.